File: 3262-find-polygon-with-the-largest-perimeter/find-polygon-with-the-largest-perimeter.java
class Solution {
    public long largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        long perimeter = 0;
        for (int num: nums){
            perimeter += num;
        }
        for (int i = nums.length - 1; i >= 2; i--) {
            if (perimeter > 2 * nums[i]) {
                return perimeter;
            }
            perimeter -= nums[i];
        }
        return -1;
    }
}

File: 3262-find-polygon-with-the-largest-perimeter/find-polygon-with-the-largest-perimeter.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()
        sum_of_sides = nums[0] + nums[1]
        max_perimeter = -1
        for i in range(2, len(nums)):
            largest_side = nums[i]
            if largest_side < sum_of_sides:
                max_perimeter = max(max_perimeter, sum_of_sides + largest_side)
            sum_of_sides += largest_side
        return max_perimeter